src/Microsoft.Diagnostics.Runtime/ClrValue.cs

```diff
@@ -0,0 +1,1077 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Runtime
+{
+    /// <summary>
+    /// A ClrValue represents the value of a field or variable.  This value may be of any type (or null).
+    /// </summary>
+    public abstract class ClrValue
+    {
+        private readonly RuntimeBase _runtime;
+
+        internal ClrValue(ClrRuntime runtime)
+        {
+            _runtime = (RuntimeBase)runtime;
+        }
+
+        internal ClrValue(RuntimeBase runtime)
+        {
+            _runtime = runtime;
+        }
+
+        /// <summary>
+        /// Returns the size of the value.
+        /// </summary>
+        public abstract int Size { get; }
+
+        /// <summary>
+        /// Returns the address of the value.
+        /// </summary>
+        public abstract ulong Address { get; }
+
+        /// <summary>
+        /// Returns the result of ReadPointer(Address) if this value is an object.
+        /// </summary>
+        protected abstract ulong Object { get; }
+
+        /// <summary>
+        /// Returns whether Address is an interior value.
+        /// </summary>
+        public abstract bool Interior { get; }
```

Comment:
Interior is tough to keep track of, but users of ClrMD already have to keep track of it to use the API now...  Also I'm ok with people pointing at random addresses to threat as a ClrValue...as at that point they likely know what they are doing.  They'd have to implement their own subclass of ClrValue though.